helmclient.go

```diff
@@ -313,7 +313,7 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 
 		o := func() error {
 			t, err := c.newTunnel()
-			if IsTillerNotFound(err) {
+			if !installTiller && IsTillerNotFound(err) {
```

Comment:
This is the fix needed for https://github.com/giantswarm/cluster-operator/pull/386. Currently we always do a permanent backoff when installing tiller. This slows the bootstrap process as we need to wait for the resync period.